qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java

```diff
@@ -58,46 +65,62 @@ public void shouldNotCompactNotExportedEvents() {
     final var logstream = clusteringRule.getLogStream(1);
     final var reader = logstream.newLogStreamReader().join();
 
-    while (getSegmentsCount(broker) <= 2) {
-      writeToLog();
-    }
+    // - write records and update the exporter position
+    ControllableExporter.updatePosition(true);
+    fillSegments(broker, SEGMENT_COUNT);
 
-    // when
+    // - write more records but don't update the exporter position
     ControllableExporter.updatePosition(false);
+    fillSegments(broker, SEGMENT_COUNT * 2);
 
-    // write more events
-    while (getSegmentsCount(broker) <= 3) {
-      writeToLog();
-    }
-    // write one more to make sure last processed position in segment 3
-    writeToLog();
-
-    // increase snapshot interval and wait
+    // - trigger a snapshot creation
     clusteringRule.getClock().addTime(SNAPSHOT_PERIOD);
     final var firstSnapshot = clusteringRule.waitForSnapshotAtBroker(broker);
 
-    // then
+    // then verify that the log still contains the records that are not exported
```

Comment:
Before this we should also verify that the compaction is completed, right? `waitUntil(getSegmentsCount() < segmentsBeforeSnapshot` . Otherwise we may not be verifying the state after compaction.

Reply:
:+1: 

> we may not be verifying the state after compaction

Reply:
Ya. You are right. But I guess there is no other way to wait for compaction.